src/mlpack/methods/logistic_regression/logistic_regression_function_impl.hpp

```diff
@@ -125,6 +125,41 @@ double LogisticRegressionFunction<MatType>::Evaluate(
     return -log(1.0 - sigmoid) + regularization;
 }
 
+/**
+ * Evaluate the logistic regression objective function given the estimated
+ * parameters for a given batch from a given point.
+ */
+template<typename MatType>
+double LogisticRegressionFunction<MatType>::Evaluate(
+                  const arma::mat& parameters,
+                  const size_t begin,
+                  const size_t batchSize) const
+{
+  // Calculating the regularization term.
+  const double regularization = 0.5 * lambda *
+      arma::dot(parameters.col(0).subvec(1, parameters.n_elem - 1),
+              parameters.col(0).subvec(1, parameters.n_elem - 1));
+
+  // Calculating the hypothesis that has to be passed to the sigmoid function.
+  const arma::vec exponents = parameters(0, 0) + predictors.t() *
```

Comment:
This is a general idea that I got, correct me if I am wrong. You are using the entire dataset in the computation here and then later on having a loop to only consider the batch we wat, that is from begin to begin + batchSize. Although we are wasting resources on the computation that is not required here. We can only consider the slice of predictors that we actually need that is predictors.cols(begin, begin + batchSize). That would save a lot of computation. What do you think?

Reply:
Oh correct, I missed it :sweat_smile: Yes, that would be a lot more efficient... I will do it :+1: